pkg/ruler/storage/instance/instance.go

```diff
@@ -284,12 +287,12 @@ func (n noopScrapeManager) Get() (*scrape.Manager, error) {
 // components cannot be reused after they are stopped so we need to recreate them
 // each run.
 func (i *Instance) initialize(_ context.Context, reg prometheus.Registerer, cfg *Config) error {
-	// explicitly set this in case this function is called multiple times
-	i.initialized = false
-
 	i.mut.Lock()
 	defer i.mut.Unlock()
 
+	// explicitly set this in case this function is called multiple times
+	i.initialized = false
```

Comment:
This is protected by the mutex below. Moved it in here so any state changes are protected